### PR TITLE
Feat: Next due date in project list

### PR DESCRIPTION
**Pull request by pbastia**
Adding the next due date in the project list, including a `None` indicator that was in the Figma wireframes

image

### CODE REVIEW

File: app/components/StatusBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    badgeColour: "#E6EBF0",
+    label: "In review",
+  },
+  customText: {
+    labelColour: "#666666",
+    badgeColour: "#E8E8E8",
+  },
+};
+
+const StatusBadge: React.FC<Props> = ({ label = "custom text", variant }) => {
```
Comment (BCerki) [inline app/components/StatusBadge.tsx:40]: What is the custom text for?

File: app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  query ProjectMilestoneDueQuery($project: ID!) @relay_test_operation {
+    query {
+      project(id: $project) {
+        ...ProjectMilestoneDue_project
+      }
+    }
+  }
+`;
+
+const mockQueryPayload = {
```
Comment (BCerki) [inline app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx:19]: Is this being used? It looks like all the tests use their own payload

File: app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      Project() {
+        return {
+          latestCommittedProjectRevision: {
+            upcomingReportingRequirementFormChange: null,
+          },
+          latestCompletedReportingRequirements: {
+            edges: [
+              {
+                node: {
+                  reportDueDate: "2020-01-02",
```
Comment (BCerki) [inline app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx:85]: Should testing dates include timestamps?

File: app/components/StatusBadge.tsx (hunk 1 above)
Comment (pbastia) [reply app/components/StatusBadge.tsx:40]: It's part of the [wireframe](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/Wireframing_Figma?node-id=1301%3A2696) for that control, so I implemented it here

File: app/components/StatusBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            fontWeight: "bold",
+          },
+        }}
+      />
+      <style jsx>
+        {`
+          .MuiChip-label {
+            color: pink;
+          }
+        `}
```
[the remark this answers is not in the document]
Comment (pbastia) [reply app/components/StatusBadge.tsx:62]: 😆  I definitely did. I'll remove it

File: app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx (hunk 3 above)
Comment (pbastia) [reply app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx:85]: eh, we're not testing the timestamp parsing function here, so for simplicity's sake I left it out

**Review by BCerki — APPROVED** [review]

**Review by BCerki — APPROVED** [review]

**Pull request merged**